### ISSUE TITLE
Internal error when declaring element LinkedList(?t) in record

### ISSUE THREAD

**Issue by ashalper-usgs** [chapel-lang/chapel#13911]
### Summary of Problem

When compiling the source on the referenced Git branch (below), the compiler stops with the following error:
```
./Structs.chpl:10: internal error: RES-CAL-NFO-0078 chpl version 1.19.0
Note: This source location is a guess.

Internal errors indicate a bug in the Chapel compiler ("It's us, not you"),
and we're sorry for the hassle.  We would appreciate your reporting this bug --
please see https://chapel-lang.org/bugs.html for instructions.  In the meantime,
the filename + line number above may be useful in working around the issue.
```
### Steps to Reproduce

**Source Code:**

Source code to reproduce the error may be found at the Git branch: https://github.com/ashalper-usgs/prms/tree/RES-CAL-NFO-0078

**Compile command:**

`chpl main.chpl`

### Configuration Information

- Output of `chpl --version`:
   `chpl version 1.19.0`
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
   ```
   CHPL_TARGET_PLATFORM: linux64
   CHPL_TARGET_COMPILER: gnu
   CHPL_TARGET_ARCH: x86_64
   CHPL_TARGET_CPU: native
   CHPL_LOCALE_MODEL: flat
   CHPL_COMM: none
   CHPL_TASKS: qthreads
   CHPL_LAUNCHER: none
   CHPL_TIMERS: generic
   CHPL_UNWIND: none
   CHPL_MEM: jemalloc
   CHPL_ATOMICS: intrinsics
   CHPL_GMP: gmp
   CHPL_HWLOC: hwloc
   CHPL_REGEXP: re2
   CHPL_AUX_FILESYS: none
   ```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
   ```
   gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)
   Copyright (C) 2015 Free Software Foundation, Inc.
   This is free software; see the source for copying conditions.  There is NO
   warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
   ```

**Comment by bradcray** [comment]
@ashalper-usgs: I have a fix for the internal error in PR #14101 by generating a user-facing "this isn't currently supported"-style error, which is what I interpret this issue as being about.  That's not to say that we shouldn't ultimately support type queries like this in fields...

**Comment by ashalper-usgs** [comment]
Very good @bradcray. This arose from me doing a too-literal translation of C to Chapel, which I shouldn't be doing anyway, so not a high priority for me personally, but thank you for making the compiler better.
